### PR TITLE
refactor: refactor starrocks, improve performance, solve bugs

### LINKED ISSUE #4376: [Bug][Task] Task in pipeline is blocked by something for 30+ mins
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I have run a pipeline that contains 4 tasks (not in parallels). They are
- jira plugin
- a private plugin to do some data transform
- starrocks plugin
- dbt plugin

The last dbt task is started at 18:19 while the previous starrocks task is finished at 17:44. I dont know what happened in this 30+ mins. 

### What do you expect to happen

The dbt tasks should run as soon as the starrocks task finished.

### How to reproduce

Dont known, seems it is occurred randomly(under some unknown conditions).

### Anything else

I queried the data from `_devlake_tasks` table, it seems there was not other tasks in that 30+ mins.  
image.  
Tell me if you need any further informations, I will provide it in my best.

### Version

v0.15.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #4377: [Bug][Task] Cancel pipeline successfully but it still running
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I have a pipeline that contains 3 tasks. They are
- jira plugin
- starrocks plugin
- dbt plugin

I want to cancel the pipeline while it is running. Using the api `DELETE /pipelines/:id`.  
The cancel api returns true but the task is still running.

### What do you expect to happen

- Should stop the running task when I call the cancel api
- Should tell me why if cancel failed

### How to reproduce

Has described above.

### Anything else

The pipeline can be cancelled when the jira task is running, but cannot when the starrocks task is running.  

How did I know the task is still running?
1. Devlake still print logs out which the starrocks plugin produce.
2. The pipeline became successful after a while.

### Version

v0.15.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #4418: [Refactor][starrocks] refactoring starrocks plugin
## What and why to refactor
refactoring starrocks plugin

## Describe the solution you'd like
1. messy code structure
2. cycle too much
3. the cursor is used indiscriminately.
4. the execution efficiency is low.
5. invalid comment

## Related issues
#4376 
#4377

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
refactor starrocks, improve performance, solve bugs.

### Does this close any open issues?
Closes #4418 
Closes #4376 
Closes #4377 

### Screenshots
export data:
image

cancel task:
image

pipeline is blocked for 30+ mins：
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/starrocks/tasks/tasks.go [hunk 1]
```diff
@@ -48,187 +49,96 @@ func (t *Table) TableName() string {
...
-						return err
-					}
-					if ok {
-						starrocksTables = append(starrocksTables, table)
-					}
-				}
-			}
-		}
+	// 2. Filter out the tables to import
+	starrocksTables, err := getImportTables(c, db)
```
Comment (klesh) [inline backend/plugins/starrocks/tasks/tasks.go:61]: import? I think it should be export?

Comment (klesh) [reply backend/plugins/starrocks/tasks/tasks.go:61]: `getExportingTables` maybe

File: backend/plugins/starrocks/tasks/tasks.go [hunk 2]
```diff
@@ -48,187 +49,96 @@ func (t *Table) TableName() string {
...
+			return errors.Convert(c.GetContext().Err())
+		default:
+		}
 		starrocksTable := strings.TrimLeft(table, "_")
 		starrocksTmpTable := fmt.Sprintf("%s_tmp", starrocksTable)
-		var columnMap map[string]string
-		var orderBy string
-		var skip bool
-		columnMap, orderBy, skip, err = createTmpTable(starrocks, db, starrocksTable, starrocksTmpTable, table, c, config)
+		columnMap, orderBy, skip, err := createTmpTable(c, starrocksDb, db, starrocksTable, starrocksTmpTable, table)
```
Comment (klesh) [inline backend/plugins/starrocks/tasks/tasks.go:80]: How about `createTmpTableInStarrocks`?

File: backend/plugins/starrocks/tasks/tasks.go [hunk 3]
```diff
@@ -48,187 +49,96 @@ func (t *Table) TableName() string {
...
-			}
-		} else {
-			return errors.NotFound.New(fmt.Sprintf("unsupported dialect %s", db.Dialect()))
-		}
-		err = errors.Convert(loadData(starrocks, c, starrocksTable, starrocksTmpTable, table, columnMap, db, config, orderBy))
-		if err != nil {
-			return errors.Convert(err)
-		}
-		err = errors.Convert(db.Exec("commit"))
+		err = putDataToDst(c, starrocksDb, db, starrocksTable, starrocksTmpTable, table, columnMap, orderBy)
```
Comment (klesh) [inline backend/plugins/starrocks/tasks/tasks.go:89]: `copyDataToDst`

File: backend/plugins/starrocks/tasks/tasks.go [hunk 4]
```diff
@@ -271,182 +181,243 @@ func createTmpTable(starrocks *sql.DB, db dal.Dal, starrocksTable string, starro
...
-	for rowsStarRocks.Next() {
-		err = rowsStarRocks.Scan(&starrocksCount)
+	req.SetBasicAuth(config.User, config.Password)
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
```
Comment (mindlesscloud) [inline backend/plugins/starrocks/tasks/tasks.go:317]: Don't forget to close the resp.Body

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**